python/ray/data/read_api.py

```diff
@@ -161,9 +164,40 @@ def read_datasource(datasource: Datasource[T],
     read_tasks = datasource.prepare_read(parallelism, **read_args)
     context = DatasetContext.get_current()
 
-    def remote_read(task: ReadTask) -> MaybeBlockPartition:
+    @ray.remote(num_cpus=0)
+    class StatsActor:
```

Comment:
One sad thing is I ended up making a StatsActor for lazy block list (for datasource reads).

Alternatively we could refactor LazyBlockList to somehow get these stats to the parent dataset, but it seems tricky.